handlers/lookup.go

```diff
@@ -64,28 +71,32 @@ func (l *lookupHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request, next
 	// receive HTTP/1.0 requests without a host headers,
 	// they optimistically add their IP address to the host header.
 	//
-	// Therefore to not expose the internal IP address of the load balancer,
+	// Therefore, to not expose the internal IP address of the load balancer,
 	// we must check that the Host is not also the source of the request.
 	//
-	// It is not vali for the Host header to be an IP address,
+	// It is not valid for the Host header to be an IP address,
 	// because Gorouter should not have an IP address as a route.
 	if r.Host == "" || hostWithoutPort(r.Host) == hostWithoutPort(r.RemoteAddr) {
+		requestInfo.RequestFailed = true
```

Comment:
I like the idea in general as it requires only one location where the flag needs to be set. I'd probably change the name to a more descriptive one like `RequestHasReachedBackend` or just `BackendReached` which in fact must start with a `false` value initially. I'd prefer this over a `the request has failed until it hasn't` kind of concept.

Reply:
@peanball I've opted for `RoundTripSuccessful` which conveys exactly what it means to say. it will only be set once we have made a successful roundtrip.

I've re-run the entire testsuite successfully. Thanks for the great hint!